### DISCUSSION TITLE
Example for link prediction validation/test time

### DISCUSSION THREAD

**Discussion by garyhsu29** [pyg-team/pytorch_geometric#5822] [Q&A] [answered]
I want to use GraphSage to train a node embedding from a huge custom dataset (~17GB) and it's in one graph. 
In my setting, there is no labeled data, so I could use link prediction to achieve my goal, but when I look into the example of GraphSage unsupervised setting https://github.com/pyg-team/pytorch_geometric/blob/master/examples/graph_sage_unsup.py
But in the example, the link prediction only applies to training time, but how could I do it during the validation/test time as well? 
I checked RandomLinkSplit but it's still not very clear to me. Does anyone have related experience?

**Comment by rusty1s** [comment, marked as answer]
What do you wanna do during evaluation? If your goal is to simply obtain node embeddings of test/val nodes, you can pass these nodes to a separate `NeighborLoader` and run your model forward procedure on these mini-batches.

**Reply by garyhsu29** [reply]
Thank you for replying! 
That says I want to also perform a linked prediction task on validation/test time.
Should I split the data before I passed to LinkNeighborLoader 
```
train_loader = LinkNeighborLoader(
    data,
    batch_size=256,
    shuffle=True,
    neg_sampling_ratio=1.0,
    num_neighbors=[10, 10],
)
```
or should I do it after the LinkNeighborLoader

**Reply by rusty1s** [reply]
You would need to create a new loader that iterates over all validation `edge_label_index`, e.g.:
```python
val_loader = LinkNeighborLoader(
    data=val_data,
    num_neighbors=[20, 10],
    edge_label_index=val_edge_label_index,
    edge_label=val_edge_label,
    batch_size=512,
    shuffle=False,
)
